src/toil/wdl/wdl_functions.py

```
@@ -235,6 +265,68 @@ def process_infile(f, fileStore):
         raise RuntimeError('Error processing file: '.format(str(f)))
 
 
+def parse_value_from_type(in_data: Any,
+                          var_type: str,
+                          read_in_file: bool = False,
+                          file_store: Optional[AbstractFileStore] = None,
+                          **kwargs):
+    """
+    Calls at runtime. This function parses and validates input from its type. File
+    import is also handled.
+
+    For values set in a task block, set `read_in_file` to True to process and read
+    all encountered files. This requires `cwd`, `temp_dir`, and `docker` to be
+    passed into this function.
+    """
+
+    def validate(val: bool, msg: str):
+        if not val:
+            raise WDLRuntimeError(f'Invalid input: {msg}')
+
+    if not in_data:  # optional type?
+        return in_data
```

We should throw an error here if the type is not in fact an optional.